Doc/library/test.rst

```diff
@@ -673,7 +1234,162 @@ The :mod:`test.support` module defines the following classes:
    On both platforms, the old value is restored by :meth:`__exit__`.
 
 
+.. class:: CleanImport(*module_names)
+
+   A context manager to force import to return a new module reference.  This
+   is useful for testing module-level behaviors, such as the emission of a
+   DeprecationWarning on import.  Example usage::
+
+      with CleanImport('foo'):
+          importlib.import_module('foo')  # New reference.
+
+
+.. class:: DirsOnSysPath(*paths)
+
+   A context manager to temporarily add directories to sys.path.
+
+   This makes a copy of :data:`sys.path`, appends any directories given
+   as positional arguments, then reverts :data:`sys.path` to the copied
+   settings when the context ends.
+
+   Note that *all* :data:`sys.path` modifications in the body of the
+   context manager, including replacement of the object,
+   will be reverted at the end of the block.
+
+
+.. class:: SaveSignals()
+
+   Class to save and restore signal handlers registered by the Python signal
+   handler.
+
+
+.. class:: Matcher()
+
+   .. method:: matches(self, d, **kwargs)
+
+      Try to match a single dict with the supplied arguments.
+
+
+   .. method:: match_value(self, k, dv, v)
+
+      Try to match a single stored value (*dv*) with a supplied value (*v*).
+
+
 .. class:: WarningsRecorder()
 
    Class used to record warnings for unit tests. See documentation of
    :func:`check_warnings` above for more details.
+
+
+.. class:: BasicTestRunner()
+
+   .. method:: run(test)
+
+      Run *test* and return the result.
+
+
+.. class:: TestHandler(logging.handlers.BufferingHandler)
```

Comment:
How to use it?

I'm not sure it should be in `test.support` rather of test_logging.